iwyu_test_util.py

```diff
@@ -45,6 +45,23 @@
 # // IWYU_ARGS: -Xiwyu --mapping_file=... -I .
 _IWYU_TEST_RUN_ARGS_RE = re.compile(r'^//\sIWYU_ARGS:\s(.*)$')
 
+# This is a line that specifies when a test should be skipped (because
+# the result would not be valid). IWYU_REQUIRES specifies a condition
+# that must be true for the test to execute. IWYU_UNSUPPORTED
+# specifies a condition in which the test is invalid, and would be
+# skipped. Multiple IWYU_REQUIRES and IWYU_UNSUPPORTED can be
+# specified, but they must be in a single comment block. For a test to
+# execute, all IWYU_REQUIRES conditions must be true, and all
+# IWYU_UNSUPPORTED conditions must be false.
+#
+# See the function _EvalCondition for the supported condition syntax.
+#
+# // IWYU_REQUIRES: cond
+# // IWYU_UNSUPPORTED: cond
+_IWYU_TEST_SKIP_RE = re.compile(r'^//\sIWYU_(REQUIRES|UNSUPPORTED):\s(.*)$')
+
+# All conditions have the same format 'LABEL ARG'
+_IWYU_COND = re.compile(r'([A-z][A-z0-9_-]+)\s+([A-z][A-z0-9_-]+)')
```

Comment:
I wonder if it would be nicer to let them read as function calls, e.g. `LABEL(ARG)`? Maybe it just gets messy.